### DISCUSSION TITLE
Is it possible to create an Auth Middleware?

### DISCUSSION THREAD

**Discussion by Dindaleon** [async-graphql/async-graphql#1186] [Q&A] [answered]
I have been playing around with extensions and prepare_request trying to get the fields from the queries, mutations, and subscriptions but I only managed to get them as a string, parse that string into something more useful, then checking if the resolver needs to be secured or not, and depending on that I verify the user token.

The issue here, for me, is to parse the query string. It's nasty! 

I wonder if someone has already created an auth extension or can somebody help me achieve this in a more efficient way without the need to parse the query string I get from the prepare_request hook.

**Comment by sunli829** [comment, marked as answer]
I think `FieldGuard` can help you

https://github.com/async-graphql/async-graphql/blob/e2e35ce93f1032d6467144c7fa8f0783e1dde04b/tests/guard.rs#L67

**Reply by Dindaleon** [reply]
> I think `FieldGuard` can help you
> 
> https://github.com/async-graphql/async-graphql/blob/e2e35ce93f1032d6467144c7fa8f0783e1dde04b/tests/guard.rs#L67

Amazing! Exactly what I wanted :)
